operator/pkg/util/util_test.go

```diff
@@ -182,3 +183,67 @@ func sameStringSlice(x, y []string) bool {
 	}
 	return len(diff) == 0
 }
+
+func TestRenderTemplate(t *testing.T) {
+	type tmplValue struct {
+		Name  string
+		Proxy string
+	}
+	tests := []struct {
+		desc     string
+		template string
+		in       tmplValue
+		want     string
+		err      error
+	}{
+		{
+			desc:     "valid-template",
+			template: "{{.Name}} uses {{.Proxy}} as sidecar",
+			in: tmplValue{
+				Name:  "istio",
+				Proxy: "envoy",
+			},
+			want: "istio uses envoy as sidecar",
+			err:  nil,
+		},
+		{
+			desc:     "empty-template",
+			template: "",
+			in: tmplValue{
+				Name:  "istio",
+				Proxy: "envoy",
+			},
+			want: "",
+			err:  nil,
+		},
+		{
+			desc:     "template with no template strings",
+			template: "this template is without handlebar expressions",
+			in: tmplValue{
+				Name:  "istio",
+				Proxy: "envoy",
+			},
+			want: "this template is without handlebar expressions",
+			err:  nil,
+		},
+		{
+			desc:     "template with missing variable",
+			template: "{{ .Name }} has replaced its control plane with {{ .Istiod }} component",
+			in: tmplValue{
+				Name:  "istio",
+				Proxy: "envoy",
+			},
+			want: "",
+			err:  errors.New(""),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			got, err := RenderTemplate(tt.template, tt.in)
+
+			if want, wantErr := tt.want, tt.err; !(got == want) || ((err == nil && wantErr != nil) || (err != nil && wantErr == nil)) {
```

Comment:
This would be a bit more understanable if you split out the err checking from the expected output checking

Reply:
@howardjohn The code is refactored to make it more understandable.